replit_river/websocket_wrapper.py

```diff
@@ -28,8 +28,8 @@ async def close(self) -> None:
         async with self.ws_lock:
             if self.ws_state == WsState.OPEN:
                 self.ws_state = WsState.CLOSING
-                task = asyncio.create_task(self.ws.close())
-                task.add_done_callback(
-                    lambda _: logger.debug("old websocket %s closed.", self.ws.id)
-                )
+                try:
+                    await self.ws.close()
```

Comment:
The ws.close() could take a long time to finish, and that's why we did a fire-and-forget here. We probably need to manage the task correctly here instead of let it be blocking.

BTW I think we actually don't need the self.ws_lock in this wrapper now, the lock in session.py is already doing its job.
Also the is_open check is not gonna race with close().

Reply:
let's try the coroutine idea that we discussed irl.